.appveyor.yml

```diff
@@ -1,86 +0,0 @@
-# AppVeyor configuration for CI Builds of ICU4C.
```

Comment:
Oh, one other thought:
I guess we can delete the `.appveyor.xml` lines from some of the build configuration files too. :)

https://github.com/daniel-ju/icu/blob/daju/port-cygwin-build-to-azure-ci/.ci-builds/.azure-valgrind.yml#L26 
https://github.com/daniel-ju/icu/blob/daju/port-cygwin-build-to-azure-ci/.ci-builds/.azure-exhaustive-tests.yml#L25

Reply:
Good point, done! :)